### PR TITLE
Add support for new JobRouter syntax (HTCONDOR-243)

### PR DESCRIPTION
**Pull request by brianhlin**
- By default, we'll continue to use `JOB_ROUTER_DEFAULTS` + `JOB_ROUTER_ENTRIES` but expose folks to the transforms and the config knobs that they'll need to use them
- For comparing pre/post transforms to the logic in the `JOB_ROUTER_DEFAULTS`, consult https://github.com/htcondor/htcondor-ce/blob/V5-branch/src/condor_ce_router_defaults#L9-L166

### CODE REVIEW

File: src/condor_ce_router_defaults [hunk 1]
```diff
@@ -6,7 +6,24 @@ import pwd
 import classad
 from collections import OrderedDict
 
-JOB_ROUTER_DEFAULTS = """JOB_ROUTER_DEFAULTS_GENERATED @=jrd
+JOB_ROUTER_CONFIG = """JOB_ROUTER_TRANSFORM_Env @=jrt
```
Comment (matyasselmeci) [inline src/condor_ce_router_defaults:9]:
```suggestion
JOB_ROUTER_CONFIG = r"""JOB_ROUTER_TRANSFORM_Env @=jrt
```
If you're gonna have backslashes in the string, you should use `r`.

File: src/condor_ce_router_defaults [hunk 2]
```diff
@@ -158,11 +175,27 @@ JOB_ROUTER_DEFAULTS = """JOB_ROUTER_DEFAULTS_GENERATED @=jrd
...
+    SET AccountingGroup    MY.AccountingGroupOSG
+@jrt
 
-    set_AccountingGroupOSG = @accounting_group@;
+JOB_ROUTER_PRE_ROUTE_TRANSFORMS = $(JOB_ROUTER_PRE_ROUTE_TRANSFORMS) AccountingGroup
+
+JOB_ROUTER_DEFAULTS_GENERATED @=jrd
+  $(JOB_ROUTER_DEFAULTS_GENERATED)
+  [
+    set_AccountingGroupOSG = {accounting_group};
```
Comment (matyasselmeci) [inline src/condor_ce_router_defaults:195]: The previous chunk is in new syntax; should this be in new syntax too? (If not, can you add a comment explaining why?)

**Review by matyasselmeci — APPROVED** [review]
A few comments but overall fine; pre-approving.

File: src/condor_ce_router_defaults (hunk 2 above)
Comment (brianhlin) [reply src/condor_ce_router_defaults:195]: Nope, the old syntax is still the default so we still need to generate `JOB_ROUTER_DEFAULTS` (which is seeded with `JOB_ROUTER_DEFAULTS_GENERATED`).

File: config/01-ce-router-defaults.conf.in [hunk 3]
```diff
@@ -103,3 +103,200 @@ SCHEDD_ATTRS = $(SCHEDD_ATTRS) HTCondorCEVersion grid_resource
...
+JOB_ROUTER_TRANSFORM_BatchRuntime @=jrt
+    # 'BatchRuntime' is in seconds but admins configure 'default_maxWallTime' and 'ROUTED_JOB_MAX_TIME' and remote
+    # submitters set 'maxWallTime' in minutes. Remote submitters set 'BatchRuntime' in minutes
+    # Set the remote batch runtime used by non-HTCondor batch systems to one of the following, in order:
+    # 1. 'maxWalltime' (seconds) from the source job
+    # 2. 'BatchRuntime' (minutes) from the source job
+    # 3. 'default_maxWallTime' (seconds) from the route
+    # 4. 'ROUTED_JOB_MAX_TIME' (minutes) from the config
+    def_walltime = 60*$(ROUTED_JOB_MAX_TIME)
+    if defined default_maxWallTime
```
[the remark this answers is not in the document]
Comment (brianhlin) [reply config/01-ce-router-defaults.conf.in]: Fixed in https://github.com/htcondor/htcondor-ce/pull/435/commits/3e795e7bd4f1e49eda591dc938e2bb70252824ea. @johnkn could you take a look at that commit to make sure I've got the syntax right?

File: src/condor_ce_router_defaults (hunk 2 above)
Comment (brianhlin) [reply src/condor_ce_router_defaults:197]: Done

**Comment by brianhlin** [comment]
@johnkn fixed in this commit https://github.com/htcondor/htcondor-ce/pull/435/commits/7993b983fca8b161b860e95cf90574e72107e49b. FWIW, though, we set `ROUTED_JOB_MAX_TIME` in the same file that we're adding this config, which the admin would have to root around to find, and if they muck with the file there's a good chance that they just break the CE completely. Another pair of suspenders never hurts, though!

**Comment by matyasselmeci** [comment]
Nothing from me.

**Comment by brianhlin** [comment]
Overriding `osg-test` failures as it requires updates to osg-test

**Pull request merged**